datasets/mc4/README.md

```diff
@@ -0,0 +1,521 @@
+| st              | Southern Sotho       |
+| su              | Sundanese            |
+| sv              | Swedish              |
+| sw              | Swahili              |
+| ta              | Tamil                |
+| te              | Telugu               |
+| tg              | Tajik                |
+| th              | Thai                 |
+| tr              | Turkish              |
+| uk              | Ukrainian            |
+| und             | Unknown language     |
+| ur              | Urdu                 |
+| uz              | Uzbek                |
+| vi              | Vietnamese           |
+| xh              | Xhosa                |
+| yi              | Yiddish              |
+| yo              | Yoruba               |
+| zh              | Chinese              |
+| zh-Latn         | Chinese (Latin)      |
+| zu              | Zulu                 |
+
+You can load the mC4 subset of any language like this:
+
+```python
+from datasets import load_dataset
+
+en_mc4 = load_dataset("mc4", "en")
+```
+
+And if you can even specify a list of languages:
+
+```python
+from datasets import load_dataset
+
+mc4_subset_with_five_languages = load_dataset("mc4", languages=["en", "fr", "es", "de", "zh"])
+```
+
+### Supported Tasks and Leaderboards
+
+mC4 is mainly intended to pretrain language models and word representations.
+
+### Languages
+
+The dataset supports 108 languages.
+
+## Dataset Structure
+
+### Data Instances
+
+An example form the `en` config is:
+
+```
+{'timestamp': '2018-06-24T01:32:39Z',
+ 'text': 'Farm Resources in Plumas County\nShow Beginning Farmer Organizations & Professionals (304)\nThere are 304 resources serving Plumas County in the following categories:\nMap of Beginning Farmer Organizations & Professionals serving Plumas County\nVictoria Fisher - Office Manager - Loyalton, CA\nAmy Lynn Rasband - UCCE Plumas-Sierra Administrative Assistant II - Quincy , CA\nShow Farm Income Opportunities Organizations & Professionals (353)\nThere are 353 resources serving Plumas County in the following categories:\nFarm Ranch And Forest Retailers (18)\nMap of Farm Income Opportunities Organizations & Professionals serving Plumas County\nWarner Valley Wildlife Area - Plumas County\nShow Farm Resources Organizations & Professionals (297)\nThere are 297 resources serving Plumas County in the following categories:\nMap of Farm Resources Organizations & Professionals serving Plumas County\nThere are 57 resources serving Plumas County in the following categories:\nMap of Organic Certification Organizations & Professionals serving Plumas County',
+ 'url': 'http://www.californialandcan.org/Plumas/Farm-Resources/'}
+```
+
+### Data Fields
+
+The data have several fields:
+
+- `url`: url of the source as a string
+- `text`: text content as a string
+- `timestamp`: timestamp as a string
+
+### Data Splits
+
+| config   | train   | validation   |
+|:---------|:--------|:-------------|
+| af       | ?       | ?            |
```

Comment:
These languages were extracted by CLD3. As @adarob explained elsewhere, CLD3 is not the best language detector there is, but it is the best compromise allowing for scale and reproducibility outside of Google. I'm mentioning this here because we immediately got questions about that, so you could head off some confusion by stating the same thing here.